src/bookmarks/background/addition.js

```diff
@@ -57,18 +59,52 @@ export async function createNewPageForBookmark(id, bookmarkInfo) {
  * @throws {Error} Error thrown if page for supplied URL not indexed.
  * @returns {Promise<void>}
  */
-export async function createBookmarkByUrl(url) {
+export async function createBookmarkByUrl(url, tabId = null) {
```

Comment:
I feel like there's too much going on in this function now with all the conditional logic. May be better to separate out into separate functions, or just another one conditionally called from this one. Not urgent though, as we can clean up more later, but if you think of a nice way feel free to implement.

Also, make sure to remove the `console.log(err)` on L74